s3tests_boto3/functional/test_iam.py

```diff
@@ -1795,3 +1796,273 @@ def test_copy_object_allow_get_src_put_dest():
+    eq(response['ResponseMetadata']['HTTPStatusCode'], 200)
+
+    response = s3_client_iam.get_bucket_versioning(Bucket=bucket)
+    eq(response['Status'], 'Suspended')
+    eq(response['ResponseMetadata']['HTTPStatusCode'], 200)
+    _delete_all_objects(s3_client_iam, bucket)
+    response = s3_client_iam.delete_bucket(Bucket=bucket)
+    eq(response['ResponseMetadata']['HTTPStatusCode'], 200)
+
+
+@attr(resource='user-policy')
+@attr(method='s3 Actions')
+@attr(operation='Verify Allow and Deny Put Bucket Versioning API using IAM policy for others')
+@attr(assertion='succeeds')
+@attr('user-policy')
+@attr('test_of_iam')
+def test_allow_deny_put_bucket_versioning_iam_policy_others():
+    client = get_iam_client()
+    s3_client_iam = get_iam_s3client()
+    bucket = get_new_bucket(client=s3_client_iam)
+    s3_client_iam.put_object(Bucket=bucket, Key='iam1buk1obj1', Body='bar')
+    s3_client_alt = get_alt_client()
+    bucket2 = get_new_bucket(client=s3_client_alt)
+    s3_client_alt.put_object(Bucket=bucket2, Key='iam2buk1obj1', Body='foobar')
+    response = s3_client_alt.put_bucket_versioning(Bucket=bucket2,
+                                                   VersioningConfiguration={"Status": "Enabled"})
+    eq(response['ResponseMetadata']['HTTPStatusCode'], 200)
+    deny_put_versioning_policy = json.dumps(
+        {
+            "Version": "2012-10-17",
+            "Statement": [
+                {
+                    "Effect": "Deny",
+                    "Action": [
+                        "s3:PutBucketVersioning"
+                    ],
+                    "Resource": [
+                        "arn:aws:s3:::iam1buk1"
+                    ]
+                }
+            ]
+        })
+    response = client.put_user_policy(PolicyDocument=deny_put_versioning_policy,
+                                      PolicyName='deny_policy_versioning',
+                                      UserName=get_alt_user_id())
+    eq(response['ResponseMetadata']['HTTPStatusCode'], 200)
+
+    e = assert_raises(ClientError, s3_client_alt.put_bucket_versioning, Bucket=bucket2,
+                      VersioningConfiguration={"Status": "Suspended"})
+    status, error_code = _get_status_and_error_code(e.response)
+    eq(status, 403)
+    eq(error_code, 'AccessDenied')
+
+    response = client.delete_user_policy(PolicyName='deny_policy_versioning',
+                                         UserName=get_alt_user_id())
+    eq(response['ResponseMetadata']['HTTPStatusCode'], 200)
+
+    allow_put_versioning_policy = json.dumps(
+        {"Version": "2012-10-17",
+         "Statement": {
+             "Effect": "Allow",
+             "Action": ["s3:PutBucketVersioning"],
+             "Resource": f"arn:aws:s3:::{bucket2}"
+         }
+         }
+    )
+    response = client.put_user_policy(PolicyDocument=allow_put_versioning_policy,
+                                      PolicyName='policy_put_versioning',
```

Comment:
You are deleting allow_policy_versioning below
```suggestion
                                      PolicyName='allow_policy_versioning',
```